vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClientChange.java

```diff
@@ -0,0 +1,170 @@
+package io.vertx.ext.mongo;
+
+import com.mongodb.client.model.changestream.ChangeStreamDocument;
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.mongo.impl.codec.json.JsonObjectCodec;
+import org.bson.BsonDocument;
+import org.bson.BsonDocumentReader;
+import org.bson.codecs.DecoderContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@DataObject
+public class MongoClientChange {
+  private static final ThreadLocal<JsonObjectCodec> codec = ThreadLocal.withInitial(() -> new JsonObjectCodec(new JsonObject()));
```

Comment:
Is there a reason for having a codec per thread?

Reply:
Nope, wasn't sure if the codec object was thread safe